db/migrate/20170207192912_encrypt_otp_secret.rb

```diff
@@ -0,0 +1,22 @@
+class EncryptOtpSecret < ActiveRecord::Migration
+  class User < ActiveRecord::Base
+  end
+
+  def up
+    encrypt_otp_secret_key
+  end
+
+  def down
+  end
+
+  def encrypt_otp_secret_key
+    User.where.not(otp_secret_key: nil).find_in_batches.with_index do |users, batch|
+      users.each do |user|
+        encrypted_attribute = EncryptedAttribute.new_from_decrypted(user.otp_secret_key)
```

Comment:
The dilemma here is that if the task is not run automatically, then the code will start breaking as soon as it is deployed. E.g. here we are switching the code to override the column called `otp_secret_key` and instead transparently encrypt/decrypt using a column called `encrypted_otp_secret_key`. Without some task or migration running, the `encrypted_otp_secret_key` column will be nil. IMO, if we turn this into a task then it makes the deployment process more complex, because there will necessarily be tasks that must be run prior to activating the new code. That's what a migration does automatically.

I've had this argument before on other projects. It's not a slam-dunk case either way IMO. I personally like to err on the side of deployments where the deploying person doesn't need to know anything special about a particular deployment or environment. The migration system takes care of the checkpointing and automation.

The issues we have had before can be mitigated by squashing migrations after every full deploy cycle. We have a problem right now where code is living too long in master before being deployed everywhere. I don't know that migration-vs-task solves that problem.